src/test/java/org/broadinstitute/consent/http/db/DacDAOTest.java

```diff
@@ -268,4 +272,49 @@ public void testFindDacsForDatasetIds() {
         assertFalse(dacs.isEmpty());
         assertEquals(1, dacs.size());
     }
+
+    @Test
+    public void testFindDatasetsAssociatedWithDac_AssignedDacId() {
+        Dataset datasetNonDac = createDataset();
+        Dac dac = createDac();
+
+        List<Dataset> results = datasetDAO.findDatasetsAssociatedWithDac(dac.getDacId());
+        assertEquals(0, results.size());
+
+        Dataset datasetAssignedDac = createDatasetWithDac(dac.getDacId());
+
+        results = datasetDAO.findDatasetsAssociatedWithDac(dac.getDacId());
+        assertEquals(1, results.size());
+        assertTrue(results.contains(datasetAssignedDac));
+    }
+
+    @Test
+    public void testFindDatasetsAssociatedWithDac_SuggestedDacId() {
+        Dataset dataset = createDataset();
+        Dac dac = createDac();
+
+        List<Dataset> results = datasetDAO.findDatasetsAssociatedWithDac(dac.getDacId());
+        assertEquals(0, results.size());
+
+        Dataset datasetAssignedDac = createDatasetWithDac(dac.getDacId());
+
+        results = datasetDAO.findDatasetsAssociatedWithDac(dac.getDacId());
+        assertEquals(1, results.size());
+        assertTrue(results.contains(datasetAssignedDac));
+
+
+        datasetDAO.insertDatasetProperties(List.of(new DatasetProperty(
+                1,
+                dataset.getDataSetId(),
+                1,
+                "dataAccessCommitteeId",
+                dac.getDacId().toString(),
+                DatasetPropertyType.Number,
+                Date.from(Instant.now()))));
+
+        results = datasetDAO.findDatasetsAssociatedWithDac(dac.getDacId());
+        assertEquals(2, results.size());
+        assertTrue(results.contains(datasetAssignedDac));
+        assertTrue(results.contains(dataset));
```

Comment:
This seems like a separate test to me. There are three tests here, no datasets found, 1 dataset found via dataset association, and 1 dataset found via property association.